### PR TITLE
Panoptes base plugin Tests

### PR DESCRIPTION
**Pull request by ilholmes**
Move TestPanoptesPluginInfo to its own file. Add tests for PanoptesPluginInfo, PanoptesPluginInfoValidators, and PanoptesBasePluginValidators and associated fixes in panoptes_base_plugin.py

### CODE REVIEW

File: tests/test_base_plugin.py [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+    def test_plugin_info_last_results(self):
+        panoptes_plugin_info = PanoptesPluginInfo("plugin_name", "path/to/plugin")
+        panoptes_plugin_info.panoptes_context = self.__panoptes_context
+        panoptes_plugin_info.kv_store_class = PanoptesTestKeyValueStore
+
+        panoptes_plugin_info.last_results = _LAST_RESULTS_TEST_VALUE
+        self.assertEqual(panoptes_plugin_info.last_results, _LAST_RESULTS_TEST_VALUE)
+
+        # Test value is cached.
+        self.assertEqual(panoptes_plugin_info.last_results, _LAST_RESULTS_TEST_VALUE)
```
Comment (varunvarma) [inline tests/test_base_plugin.py:183]: We probably need to mock and fail the Redis connection and ensure that this continues to work.

File: tests/test_base_plugin.py [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+def mock_metadata_kv_store():
+    return MagicMock(side_effect=Exception)
+
+
+class TestPanoptesPluginInfo(unittest.TestCase):
+    @patch('redis.StrictRedis', panoptes_mock_redis_strict_client)
+    @patch('kazoo.client.KazooClient', panoptes_mock_kazoo_client)
+    def setUp(self):
+        self.my_dir, self.panoptes_test_conf_file = _get_test_conf_file()
+        self.__panoptes_context = PanoptesContext(self.panoptes_test_conf_file,
```
Comment (varunvarma) [inline tests/test_base_plugin.py]: Dunder names for properties is a bad pattern I started - let's not propagate it. Let's use a single underscore instead.

File: tests/test_base_plugin.py [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+        with patch('yahoo_panoptes.framework.plugins.panoptes_base_plugin.PanoptesPluginInfo.configMtime',
+                   mock_configMtime):
+            with patch('yahoo_panoptes.framework.plugins.panoptes_base_plugin.PanoptesPluginInfo.moduleMtime',
+                       mock_moduleMtime):
+                # Ensure first if-block in execute_now returns False
+                panoptes_plugin_info.last_results = int(_TIMESTAMP)
+                panoptes_plugin_info.last_executed = int(_TIMESTAMP)
+                self.assertFalse(panoptes_plugin_info.execute_now)
+
+    def test_plugin_info_execute_now_2(self):
```
Comment (varunvarma) [inline tests/test_base_plugin.py]: This seems like a duplicate of test_plugin_info_execute_now

File: tests/test_base_plugin.py [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+    def test_plugin_info_execute_now_3(self):
+        panoptes_plugin_info = PanoptesPluginInfo("plugin_name", "path/to/plugin")
+        panoptes_plugin_info.panoptes_context = self.__panoptes_context
+        panoptes_plugin_info.kv_store_class = PanoptesTestKeyValueStore
+        panoptes_plugin_info.config_filename = "tests/plugins/polling/test/plugin_polling_test.panoptes-plugin.example"
+        panoptes_plugin_info.details.read(panoptes_plugin_info.config_filename)
+
+        mock_moduleMtime = _TIMESTAMP - 1
+        mock_configMtime = _TIMESTAMP - 2
+        with patch('yahoo_panoptes.framework.plugins.panoptes_base_plugin.PanoptesPluginInfo.configMtime',
```
Comment (varunvarma) [inline tests/test_base_plugin.py]: How about we merge this with  test_plugin_info_execute_now since we can avoid the duplicate setup code

**Review by varunvarma — CHANGES_REQUESTED** [review]

File: tests/test_base_plugin.py [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+
+        panoptes_plugin_info_2 = PanoptesPluginInfo("plugin_name", "path/to/plugin")
+        panoptes_plugin_info_2.panoptes_context = self.__panoptes_context
+
+        #  Patch timeout to speed up test
+        with patch('yahoo_panoptes.framework.plugins.panoptes_base_plugin.const.PLUGIN_AGENT_LOCK_ACQUIRE_TIMEOUT', 1):
+            self.assertFalse(panoptes_plugin_info_2.lock.locked)
+
+
+def _get_test_conf_file():
```
Comment (varunvarma) [inline tests/test_base_plugin.py]: Lets move this to helpers.py in the top level test directory

**Review by varunvarma — CHANGES_REQUESTED** [review]

File: tests/test_base_plugin.py (hunk 1 above)
Comment (ilholmes) [reply tests/test_base_plugin.py:183]: Moved lines 189, 190 (and 205, 206) inside the context manager to make this test clearer.

**Pull request merged**